### PR TITLE
test(frontend): scope-correct ActiveWatchlist tests (not in spec #250)

### LINKED ISSUE #312: test(frontend): scope-correct ActiveWatchlist tests (not in spec #250)
Scope spillover from #250. ActiveWatchlist.test.tsx added OOS during coverage ratchet — not in spec or priority list. Retained because excision breaks coverage thresholds. Automatically triaged by MarketHawk Dark Factory scope enforcement.

### PR DESCRIPTION
**Pull request by omniscient**
Closes #312

## Summary
Automated implementation for issue #312.

## Preview
_No preview environment — this change does not affect the running app ('All changes are test files (.test.tsx), documentation (docs/codeindex-hotspots.md), and agent workflow config (.archon/memory/). No runtime code changes. No backend/app, frontend/src/pages/<impl>, package.json, Dockerfile, or seed changes.')._

## Commands
```bash
# Iterate after feedback
docker compose --profile factory run --rm dark-factory "Continue issue #312"

# Tear down preview when done
docker compose --profile factory run --rm dark-factory "Close issue #312"
```

---
*Generated by MarketHawk Dark Factory*

### CODE REVIEW

**Review by omniscient — COMMENTED** [review]
## 🏭 Dark Factory Code Review

**0** blocking · **4** advisory finding(s).

### Findings not anchorable to the diff
- [low] naming `frontend/src/pages/ActiveWatchlist/ActiveWatchlist.test.tsx:53` — at-limit `getByText('50')` count assertion is brittle against the `/ 50` denominator text; prefer asserting the colored count span directly (e.g. by its `text-red-400` class) to avoid a future false match.

File: frontend/src/pages/ActiveWatchlist/WatchlistTable.test.tsx [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+function renderTable(
+  items: WatchlistItem[],
+  liveData: Record<string, SymbolLiveData> = {}
+) {
+  return render(
+    <MemoryRouter>
+      <WatchlistTable items={items} liveData={liveData} />
+    </MemoryRouter>
+  );
+}
```
Comment (omniscient) [inline frontend/src/pages/ActiveWatchlist/WatchlistTable.test.tsx:54]: **edge-case** · **low** — no-live-data dash test asserts only `getAllByText('—').length > 0`, which passes whether the dash came from price, session, or notes; tighten to the price cell.

File: frontend/src/pages/ActiveWatchlist/WatchlistTable.test.tsx [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    const { container } = renderTable(
+      [makeItem()],
+      { AAPL: makeLive({ lastTickAt: Date.now() - 20_000 }) }
+    );
+    expect(container.querySelector('span.text-gray-500.font-semibold')).not.toBeNull();
+  });
+
+  it('applies text-positive to positive priceChangePct', () => {
+    const { container } = renderTable([makeItem()], { AAPL: makeLive({ priceChangePct: 2.5 }) });
+    expect(container.querySelector('span.text-positive')).not.toBeNull();
```
Comment (omniscient) [inline frontend/src/pages/ActiveWatchlist/WatchlistTable.test.tsx:78]: **maintainability** · **low** — `text-positive` positive-pct test is satisfied by SessionCell's REG color too (default session `regular`); use `session: 'closed'` or scope the query to PriceCell so it isolates the cell under test.

File: frontend/src/pages/ActiveWatchlist/WatchlistTable.test.tsx [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+    const { container } = renderTable([makeItem()], { AAPL: makeLive({ session: 'pre' }) });
+    expect(screen.getByText('PRE')).toBeInTheDocument();
+    expect(container.querySelector('span.text-yellow-400')).not.toBeNull();
+  });
+
+  it('shows REG label with text-positive for regular session', () => {
+    const { container } = renderTable([makeItem()], { AAPL: makeLive({ session: 'regular' }) });
+    expect(screen.getByText('REG')).toBeInTheDocument();
+    expect(container.querySelector('span.text-positive')).not.toBeNull();
+  });
```
Comment (omniscient) [inline frontend/src/pages/ActiveWatchlist/WatchlistTable.test.tsx:108]: **maintainability** · **low** — REG `text-positive` assertion can match the PriceCell pct span instead of SessionCell; query within the session cell to make the assertion specific.
